fabric8/src/main/java/org/jboss/arquillian/ce/fabric8/F8OpenShiftAdapter.java

```diff
@@ -487,14 +493,56 @@ private Object triggerBuild(String namespace, String buildName, String secret, S
     }
 
     protected OpenShiftResourceHandle createResourceFromStream(InputStream stream) throws IOException {
-        ModelNode json;
-        try {
-            json = ModelNode.fromJSONStream(stream);
+    	
+    	try {
+	    	StringBuilder textBuilder = new StringBuilder();
+	        try (Reader reader = new BufferedReader(new InputStreamReader
+	          (stream, Charset.forName(StandardCharsets.UTF_8.name())))) {
+	            int c = 0;
+	            while ((c = reader.read()) != -1) {
+	                textBuilder.append((char) c);
+	            }
+	        }
+	        String content = textBuilder.toString();
+	       
+	        char nonwhitespace;
+	        int index=0;
+	        do {
+	        	nonwhitespace=content.charAt(index++);
+	        } while (index < content.length() && Character.isWhitespace(nonwhitespace));
+	        
+	        if ( nonwhitespace == '{' ) {
+	        	ModelNode json;
+	            json = ModelNode.fromJSONString(content);
+	            String kind = json.get("kind").asString();
+	            return createResourceFromJson(kind, json);
+	        } else {
+	        	StringTokenizer tokenizer = new StringTokenizer(content.trim(),":\n");
+	        	tokenizer.nextToken();
+	        	String kind = tokenizer.nextToken().trim();
+	        	
+	        	return createResourceFromYaml(kind, content);
+	        }
+	        
         } finally {
```

Comment:
Given those createResourceFromXXX have quite the same implementation, I'd suggest:
 - Stick with only one method, rename it to createResourceFromString(kind, content)
 - Extract the content before calling it and passing it as the second argument

At this point I wonder if it's worth to write that logic to find out the file type (JSON or YAML) or if it just makes sense to pass the entire file (as the second argument, as a string) to createResourceFromString()...